File: wasm/bmxt-core/src/cmd/notes.rs
use crate::meta::Cmd;
use crate::model::{DispatchJson, Effect};

pub const CMD: Cmd = Cmd {
    name: "notes",
    aliases: &[],
    usage_primary: "notes",
};

pub fn run(args: &[String]) -> DispatchJson {
    match args.len() {
        0 | 1 => DispatchJson::effects(vec![Effect::ReleaseNotesCurrent]),
        2 => {
            let a = args[1].as_str();
            if a == "--list" || a.eq_ignore_ascii_case("-l") {
                return DispatchJson::effects(vec![Effect::ReleaseNotesList]);
            }
            if a.starts_with('-') {
                return usage();
            }
            let version = args[1].clone();
            DispatchJson::effects(vec![Effect::ReleaseNotesVersion { version }])
        }
        _ => usage(),
    }
}

fn usage() -> DispatchJson {
    DispatchJson::lines(vec![
        "usage: notes              — release notes for the current extension version".to_string(),
        "       notes <version>    — notes for that version key (e.g. 0.0.8)".to_string(),
        "       notes --list | -l  — list versions that have entries".to_string(),
    ])
}

File: lib/features/bmxt-core/cmd/notes.ts
import type { CmdMeta } from "../types"
import { effectsDispatch, linesDispatch } from "../types"

export const CMD: CmdMeta = {
  name: "notes",
  aliases: [],
  usagePrimary: "notes"
}

function usage() {
  return linesDispatch([
    "usage: notes              — release notes for the current extension version",
    "       notes <version>    — notes for that version key (e.g. 0.0.8)",
    "       notes --list | -l  — list versions that have entries"
  ])
}

export function run(args: string[]) {
  if (args.length <= 1) {
    return effectsDispatch([{ kind: "release_notes_current" }])
  }
  if (args.length === 2) {
    const a = args[1]
    if (a === "--list" || a.toLowerCase() === "-l") {
      return effectsDispatch([{ kind: "release_notes_list" }])
    }
    if (a.startsWith("-")) {
      return usage()
    }
    return effectsDispatch([{ kind: "release_notes_version", version: args[1] }])
  }
  return usage()
}
